tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -630,38 +630,30 @@ internal bool IsMetaStructSupported()
             return _spanMetaStructs;
         }
 
-        internal void UpdateActiveAddresses()
+        private void UpdateActiveAddresses()
         {
             // So far, RASP is the only one that uses this
-            if (_settings.RaspEnabled && _waf?.IsKnowAddressesSuported() == true)
+            if (_waf?.IsKnowAddressesSuported() is true)
             {
                 var addresses = _waf.GetKnownAddresses();
                 Log.Debug("Updating WAF active addresses to {Addresses}", addresses);
-                _activeAddresses = addresses is null ? null : new HashSet<string>(addresses);
+                _activeAddresses = [..addresses];
             }
             else
             {
                 _activeAddresses = null;
             }
         }
 
-        internal bool AddressEnabled(string address)
+        public bool AddressEnabled(string address)
```

Comment:
NIT: Is public required?

Reply:
yes as it's now mocked in the unit tests `ContextUserEventTests` , cant seem to find another way 🤔

Reply:
> yes as it's now mocked in the unit tests ContextUserEventTests , cant seem to find another way

IMO, requiring mocking like this is a strong indication that we need an additional layer of abstraction and/or dependency injection.